### PR TITLE
nncp: 5.3.3 -> 6.2.0

### PR DESCRIPTION
**Pull request by woffs**
###### Motivation for this change
- upstream changed build system from make to redo

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/nncp/default.nix [hunk 1]
```diff
@@ -9,24 +10,28 @@
...
+    redo ''${enableParallelBuilding:+-j''${NIX_BUILD_CORES}}
   '';
 
-  makeFlags = [
-    "PREFIX=${placeholder "out"}"
-    "CFGPATH=/etc/nncp.hjson"
-    "SENDMAIL=/run/wrappers/bin/sendmail"
-  ];
+  installPhase = ''
+    export PREFIX=${placeholder "out"}
```
Comment (alyssais) [inline pkgs/tools/misc/nncp/default.nix:30]: No need for `placeholder` here.

Either

```sh
export PREFIX=$out
```

Or

```nix
  PREFIX = placeholder "out";
```

(Either is acceptable. I personally prefer the former.)

Comment (woffs) [reply pkgs/tools/misc/nncp/default.nix]: Thank you. I've changed this.

File: pkgs/tools/misc/nncp/default.nix [hunk 2]
```diff
@@ -9,24 +10,28 @@
...
   };
 
-  nativeBuildInputs = [ go ];
+  nativeBuildInputs = [ go redo-apenwarr ];
 
-  preConfigure = ''
+  buildPhase = ''
     export GOCACHE=$PWD/.cache
+    export CFGPATH=/etc/nncp.hjson
+    export SENDMAIL=/run/wrappers/bin/sendmail
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/nncp/default.nix:25]: Shouldn't this point to the nix version or am I missing something?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99304` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nncp</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99304` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nncp</li>
  </ul>
</details>

File: pkgs/tools/misc/nncp/default.nix (hunk 2 above)
Comment (woffs) [reply pkgs/tools/misc/nncp/default.nix:25]: It's only a default value, not a dependency. Do you have a better suggestion? What would be the nix version, without adding a dependency?

**Comment by woffs** [comment]
> redo: install: exists and not marked as generated; not redoing.

Ah, case insensitive file system breaks this :-(

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99304` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nncp</li>
  </ul>
</details>

File: pkgs/tools/misc/nncp/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/nncp/default.nix]: You can't add a default enabled value without adding a depedency.

File: pkgs/tools/misc/nncp/default.nix [hunk 3]
```diff
@@ -9,24 +10,29 @@
...
   };
 
-  nativeBuildInputs = [ go ];
+  nativeBuildInputs = [ go redo-apenwarr ];
 
-  preConfigure = ''
+  buildPhase = ''
     export GOCACHE=$PWD/.cache
+    export CFGPATH=/etc/nncp.hjson
+    export SENDMAIL=/run/wrappers/bin/sendmail # just a default value for generated config file
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/nncp/default.nix:25]: This won't work on NixOS and adding defaults which are broken is not great.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99304` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nncp</li>
  </ul>
</details>

File: pkgs/tools/misc/nncp/default.nix (hunk 2 above)
Comment (woffs) [reply pkgs/tools/misc/nncp/default.nix]: what's the solution?
- keep upstream /usr/sbin/sendmail, or
- set it to /run/wrappers/bin/sendmail which only works on NixOS, or
- pull in a not-needed and maybe bloated and unconfigured MTA dependency?

Also I must note this is not the main topic of this PR and the path is already there.

File: pkgs/tools/misc/nncp/default.nix (hunk 3 above)
Comment (woffs) [reply pkgs/tools/misc/nncp/default.nix:25]: see https://github.com/NixOS/nixpkgs/pull/99304#discussion_r530870941

Comment (woffs) [reply pkgs/tools/misc/nncp/default.nix]: I changed it to plain "sendmail", which is portable. And sending mail is optional and not a dependency. The generated config file is to be edited by the user anyway.

File: pkgs/tools/misc/nncp/default.nix [hunk 4]
```diff
@@ -9,24 +10,29 @@
...
 
-  nativeBuildInputs = [ go ];
+  nativeBuildInputs = [ go redo-apenwarr ];
 
-  preConfigure = ''
+  buildPhase = ''
     export GOCACHE=$PWD/.cache
+    export CFGPATH=/etc/nncp.hjson
+    export SENDMAIL=sendmail # default value for generated config file
+    redo ''${enableParallelBuilding:+-j''${NIX_BUILD_CORES}}
```
Comment (doronbehar) [inline pkgs/tools/misc/nncp/default.nix:26]: Hooks https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md .

File: pkgs/tools/misc/nncp/default.nix [hunk 5]
```diff
@@ -9,24 +10,29 @@
...
 
-  makeFlags = [
-    "PREFIX=${placeholder "out"}"
-    "CFGPATH=/etc/nncp.hjson"
-    "SENDMAIL=/run/wrappers/bin/sendmail"
-  ];
+  installPhase = ''
+    export PREFIX=$out
+    rm -f INSTALL # work around case insensitivity
+    redo install
```
Comment (doronbehar) [inline pkgs/tools/misc/nncp/default.nix:32]: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md

**Review by doronbehar — COMMENTED** [review]
Otherwise LGTM.

File: pkgs/tools/misc/nncp/default.nix (hunk 4 above)
Comment (woffs) [reply pkgs/tools/misc/nncp/default.nix:27]: done.

**Pull request merged**